presto-main/src/main/java/com/facebook/presto/type/TypeRegistry.java

```diff
@@ -216,7 +226,7 @@ private static boolean canCastTypeBase(String fromTypeBase, String toTypeBase)
         }
         switch (fromTypeBase) {
             case StandardTypes.BIGINT:
```

Comment:
you should not change `parametersMatch` to `true` within the loop. Initial value should be set to `true` and then the loop should only change it to `false` if one of parameters does not match.